editor/post-title/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+function PostTitle( { title, onUpdate } ) {
+	const onChange = ( event ) => onUpdate( event.target.value );
+
+	return (
+		<h1 className="editor-post-title">
+			<input
+				className="editor-post-title__input"
+				type="text"
+				value={ title }
+				onChange={ onChange }
+				placeholder={ wp.i18n.__( 'Write title' ) }
+			/>
+		</h1>
+	);
+}
+
+export default connect(
+	( state ) => ( {
+		title: state.editor.edits.title === undefined
+			? state.currentPost.title.raw
+			: state.editor.edits.title,
+	} ),
+	( dispatch ) => {
+		return {
+			onUpdate( title ) {
+				dispatch( {
+					type: 'EDIT_POST',
+					post: { title }
```

Comment:
Not specific to this pull request, but I'm wondering if the `post` key naming here is misleading in how we're only applying changes as a patch, not as a full post object. 

@nylen @youknowriad do you have any thoughts on this? Would another name like `edits`, `values`, `properties` be any clearer?